File: templates/problems/17_math_geometry/07_multiply_strings/multiply_strings_test.go
package multiply_strings

import "testing"

func TestMultiply(t *testing.T) {
	tests := []struct {
		name     string
		num1     string
		num2     string
		expected string
	}{
		{"example 1", "2", "3", "6"},
		{"example 2", "123", "456", "56088"},
		{"multiply by zero", "0", "52", "0"},
		{"both zeros", "0", "0", "0"},
		{"single digits", "9", "9", "81"},
		{"large numbers", "999", "999", "998001"},
		{"one and number", "1", "12345", "12345"},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := multiply(tt.num1, tt.num2)
			if got != tt.expected {
				t.Errorf("multiply(%q, %q) = %q, want %q", tt.num1, tt.num2, got, tt.expected)
			}
		})
	}
}

File: templates/problems/17_math_geometry/07_multiply_strings/typescript/multiplyStrings.test.ts
import { describe, it, expect } from 'vitest';
import { multiply } from './multiplyStrings';

describe('MultiplyStrings', () => {
    it('example 1', () => {
        expect(multiply("2", "3")).toBe("6");
    });

    it('example 2', () => {
        expect(multiply("123", "456")).toBe("56088");
    });

    it('multiply by zero', () => {
        expect(multiply("0", "52")).toBe("0");
    });

    it('both zeros', () => {
        expect(multiply("0", "0")).toBe("0");
    });

    it('single digits', () => {
        expect(multiply("9", "9")).toBe("81");
    });

    it('large numbers', () => {
        expect(multiply("999", "999")).toBe("998001");
    });

    it('one and number', () => {
        expect(multiply("1", "12345")).toBe("12345");
    });

    it('overflow case', () => {
        expect(multiply("498828660196", "840477629533")).toBe("419254329864656431168468");
    });
});
